src/PreallocatedLinearOperators.jl

```diff
@@ -0,0 +1,135 @@
+export PreallocatedLinearOperator
+
+abstract type PreallocatedAbstractLinearOperator{T,F1,F2,F3} <: AbstractLinearOperator{T,F1,F2,F3} end
+
+"""
+Type to represent a linear operator with preallocation. Implicit modifications may
```

Comment:
Maybe this docstring should describe the behavior of such an operator a little bit, e.g., the result of the operation `A * x` is placed in a preallocated vector stored inside the operator. Thus, this result is overwritten if another `A * y` is performed. Something like that?!